rest-api-spec/test/template/30_validate_search_template.yaml

```diff
@@ -0,0 +1,55 @@
+---
+"Indexed Template validate tests":
+
+  - do:
+      put_template:
+        id: "1"
+        body: { "template": { "query": { "match": { "text": "{{my_value}}" } }, "size": "{{my_size}}" } }
+  - match: { _id: "1" }
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      validate_search_template:
+        body: { "id": "1", "params": { "my_value": "foo", "my_size": 20 } }
+
+  - match: { template_output.query.match.text: "foo" }
+  - match: { template_output.size: "20" }
+
+  - do:
+      validate_search_template:
+        body: { "id": "1", "params": { "my_value": "bar", "my_size": 100 } }
+
+  - match: { template_output.query.match.text: "bar" }
+  - match: { template_output.size: "100" }
```

Comment:
> This is a known bug and something we need to fix separately.

I don't think it is a bug, rather just a limitation.  If we need to pass in a number as a mustache variable then it should be without quotes, which makes it invalid JSON, which means that the template needs to be an escaped string instead of a JSON object.